Packs/MicrosoftTeams/Integrations/MicrosoftTeams/README.md

```diff
@@ -1,74 +1,245 @@
+
+1. *Client Credentials Flow*
+2. *Authorization Code Flow* (Supports the [chat commands](#chat-commands))
+
+**Perform the following steps to add the needed permissions**:
+
+1. Go to your [Microsoft Azure portal](https://portal.azure.com/), and from the left navigation pane select **Azure Active Directory > App registrations**.
+2. Search for and click **Demisto Bot**.
+3. Click **API permissions > Add a permission > Microsoft Graph > Application/Delegated permissions**.
+4. For each of the next permissions listed below, search for the permission, select the checkbox, and click **Add permissions**.
+
+Note ⚠️: The [microsoft-teams-ring-user](https://learn.microsoft.com/en-us/graph/api/application-post-calls?view=graph-rest-1.0&tabs=http) command requires authenticating with `Client Credentials` due to a limitation in Microsoft's permissions system. (Calling this command will perform the authentication seemlessly)
+
+#### For Client Credentials Flow add the following
+
+**Application Permissions**:
+
+- `User.Read.All`
+- `GroupMember.Read.All`
+- `Channel.ReadBasic.All`
+- `ChannelMember.ReadWrite.All`
+- `Channel.Create`
+- `Channel.Delete.All`
+- `OnlineMeetings.ReadWrite.All`
+- `Calls.Initiate.All`
+
+#### For Authorization Code Flow add the following
+
+**Delegated Permissions**:
+
+- `User.Read.All`
+- `GroupMember.Read.All`
+- `Channel.ReadBasic.All`
+- `ChannelMember.ReadWrite.All`
+- `Channel.Create`
+- `Channel.Delete.All`
+- `ChannelMessage.Send`
+- `OnlineMeetings.ReadWrite.All`
+- `Chat.ReadWrite`
+- `AppCatalog.Read.All`
+- `TeamsAppInstallation.ReadWriteSelfForChat`
+
+**Application permissions:** (For `microsoft-teams-ring-user`)
+- `User.Read.All`
+- `Calls.Initiate.All`
+
+Alternatively, you can check each relevant command section below for the minimum permissions it requires.
+
+5. Verify that all permissions were added, and click **Grant admin consent**.
+
+6. When prompted to verify granting permissions, click **Yes**, and verify that permissions were successfully added.
+
+7. Click **Authentication** > **Platform configurations** > **Add a platform**. Choose **Web** and add Redirect URIs: <https://oproxy.demisto.ninja/authcode>
+
+### 3. Configure Microsoft Teams on Cortex XSOAR/XSIAM
+
+There are two flows in order to configure the integration as was mentioned before:
+Please note that you need to use the flow to which you have added permissions.
+
+1. Client Credentials
+2. Authorization Code (Support [The chat commands](#chat-commands))
+
+For more detailed instructions, refer to [the relevant authentication flow](#authentication-using-the-authorization-code-flow) section.
 
-## First time setup
+In order to configure the integration follow the next steps:
 
-Refer to the [Setup Video](#setup-video) and the [Prerequisites](#prerequisites) sections for detailed steps to configure the teams bot and instance for the first time.
+1. Search for Microsoft Teams integration after you have installed it from the marketplace.
```

Comment:
Search for Microsoft Teams integration after you have installed it from the Marketplace.